File: android_aidl/app/src/main/java/top/zcwfeng/aidl/MainActivity.java
package top.zcwfeng.aidl;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    // service 端aidl client 参见 android_kotlin 项目demo
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        startService(new Intent(this,ZcwService.class));
    }
}

File: android_hilt/app/src/main/java/top/zcwfeng/hilt/MainActivity.kt
package top.zcwfeng.hilt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import dagger.hilt.android.AndroidEntryPoint
import top.zcwfeng.hilt.databinding.ActivityMainBinding
import top.zcwfeng.hilt.model.User
import javax.inject.Inject

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    @Inject
    lateinit var user: User
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val databinding: ActivityMainBinding =
            DataBindingUtil.setContentView(this, R.layout.activity_main)
        databinding.content.text = "${user.name}现在的心情是${user.desc}"

    }
}